shell/platform/common/cpp/client_wrapper/include/flutter/event_sink.h

```diff
@@ -19,28 +19,49 @@ class EventSink {
   EventSink(EventSink const&) = delete;
   EventSink& operator=(EventSink const&) = delete;
 
+  // DEPRECATED. Use the reference version below. This will be removed in the
+  // near future.
+  void Success(const T* event) { SuccessInternal(event); }
+
+  // Consumes a successful event
+  void Success(const T& event) { SuccessInternal&(event); }
+
   // Consumes a successful event.
-  void Success(T* event = nullptr) { SuccessInternal(event); }
+  void Success() { SuccessInternal(nullptr); }
 
-  // Consumes an error event.
+  // DEPRECATED. Use the reference version below. This will be removed in the
```

Comment:
> Probably not, since warnings-are-errors

Exactly. I tried that the last time I did one of these and realized it would just make it an immediate breaking change, which is unfortunate.

In practice, since Windows is still subject to change without warning, it's not a big deal (I'm doing it in stages so I can not break FDE and google3 without having a chance to pre-fix them), but long term it would be nice if we could use them to do soft breaking changes. It's a potential argument for relaxing the template to not treat warnings as errors.